dist/stat_data.py

```diff
@@ -244,6 +244,9 @@ def __init__(self, name, desc, flags=''):
     BlockCacheStat('block_prefetch_pages_queued', 'pre-fetch pages queued'),
     BlockCacheStat('block_prefetch_pages_read', 'pre-fetch pages read in background'),
     BlockCacheStat('block_prefetch_skipped', 'pre-fetch not triggered by page read'),
+    BlockCacheStat('block_prefetch_skipped_disk_read_count', 'pre-fetch not triggered for disk read count'),
```

Comment:
Thanks for splitting these out. These categories make sense. I have some wording nitpicks - we should say "... triggered due to disk ..." rather than "for" - it sounds like we're trying to prefetch a disk read count at the moment, which wouldn't make sense. Same with internal session.

The "same ref" one is a little confusing - it sounds like that's a problem, but really we're just saying we've recently done a prefetch for this ref, and we're not going to do another until the counter ticks down. I don't know how to clearly express that in a one-line stat description, possibly "prefetch not repeating for recently prefetched ref"? Happy to chat offline about this.

Reply:
Yes, I'll have a think about that name as well...